src/walletSdk/Horizon/transaction/TransactionBuilder.ts

```diff
@@ -82,4 +114,27 @@ export class TransactionBuilder {
     });
     return this.builder.build();
   }
+
+  transferWithdrawalTransaction(
+    transaction: WithdrawTransaction,
+    assetId: StellarAssetId,
+  ): TransactionBuilder {
+    if (transaction.status !== TransactionStatus.pending_user_transfer_start) {
+      throw new WithdrawalTxNotPendingUserTransferStartError(
+        transaction.status,
+      );
+    }
+
+    if (!(transaction.withdraw_memo_type && transaction.withdraw_memo)) {
+      throw new WithdrawalTxMissingMemoError();
```

Comment:
should we add errors for missing asset and amount fields as well?

Reply:
unless the `transfer()` function already throw errors for those missing fields 🤔

Reply:
ya if those are empty creating the payment operation in `transfer` will throw an error, so I don't think we need to add the checks